code/datums/components/bayonet_attachable.dm

```diff
@@ -0,0 +1,212 @@
+/**
+ * Component which allows you to attach a bayonet to an item,
+ * be it a piece of clothing or a tool.
+ */
+/datum/component/bayonet_attachable
+	/// Whenever we can remove the bayonet with a screwdriver
+	var/removable = TRUE
+	/// If passed, we wil simply update our item's icon_state when a bayonet is attached.
+	/// Formatted as parent_base_state-[bayonet_icon_state-state]
+	var/bayonet_icon_state
+	/// If passed, we will use a specific overlay instead of using the knife itself
+	/// The state to take from the bayonet overlay icon if supplied.
+	var/bayonet_overlay
+	/// This is the icon file it grabs the overlay from.
+	var/bayonet_overlay_icon
+	/// Offsets for the bayonet overlay
+	var/offset_x = 0
+	var/offset_y = 0
+	/// If this component allows sawing off the parent gun/should be deleted when the parent gun is sawn off
+	var/allow_sawnoff = FALSE
+
+	// Internal vars
+	/// Currently attached bayonet
+	var/obj/item/knife/bayonet
```

Comment:
Any reason to typecast this to knife? Can't see that we use any knife-specific properties.